File: 04_linear_search/zig/linear_search.zig
const std = @import("std");

pub fn linearSearch(arr: []const i32, target: i32) !?usize {
    var i: usize = 0;
    while (i < arr.len) : (i += 1) {
        if (arr[i] == target) {
            return i; // Return the index if the target is found
        }
    }
    return null; // Return null if the element is not found
}

pub fn main() !void {
    const arr: []const i32 = &[_]i32{3, 45, 1, 2, 99}; // Use fixed-size array initialization
    const target: i32 = 1;
    const result = try linearSearch(arr, target) orelse {
        // Handle the case where the element is not found
        std.debug.print("Element not found in the array\n", .{});
        return;
    };
    // If we reach this point, the element was found
    std.debug.print("Element found at index: {}\n", .{result});
}

File: 04_linear_search/go/linear_search.go
package main

import "fmt"

func linearSearch(arr []int, target int) int {
	for i, value := range arr {
		if value == target {
			return i
		}
	}
	return -1
}

func main() {
	arr := []int{1, 3, 5, 7, 9}
	target := 7
	result := linearSearch(arr, target)
	if result != -1 {
		fmt.Printf("Element found at index %d\n", result)
	} else {
		fmt.Println("element not found in the array")
	}
}